src/main/java/spoon/reflect/visitor/ImportConflictDetector.java

```diff
@@ -196,6 +198,18 @@ protected void handleTypeReference(CtTypeReference<?> ref, LexicalScope nameScop
 				ref.setSimplyQualified(false);
 				return false;
 			});
+
+			if (!ref.isImplicit() && ref.isSimplyQualified()) {
+				Map<String, String> encounteredNames = ((LexicalScopeScanner) scanner).getEncounteredImportedQualifiedNames();
+				if (!encounteredNames.containsKey(ref.getSimpleName())) {
+					encounteredNames.putIfAbsent(ref.getSimpleName(), ref.getQualifiedName());
+				}
```

Comment:
Shouldn't the `putIfAbsent` replace the check in the if?

Reply:
That is true, I refactored the check a few times and it became redundant :)